### PR TITLE
feature: add `divide` subcommand

### LINKED ISSUE #89: Add the ability to divide a CIDR range into smaller ranges
The CIDR range division feature allow users to easily divide a given CIDR range into a specified number of smaller subnets.

**Why**: this enhancement is crucial for users dealing with IP address management, allowing them to efficiently and accurately divide CIDR ranges as needed. Improving this feature will enhance the overall usability of `cidr`.

**Example:**

Introduce a `divide` subcommand, where you specify both a CIDR range as well as an integer to divide it respectively.

```
$ cidr divide 10.0.0.0/16 9
10.0.0.0/20
10.0.16.0/20
10.0.32.0/20
10.0.48.0/20
10.0.64.0/20
10.0.80.0/20
10.0.96.0/20
10.0.112.0/20
10.0.128.0/20
```

h/t @steynovich for proposing this enhancement to cidr in a 1-on-1.

### PR DESCRIPTION
**Pull request by Phaze228**
… network command

## What
* Rewrote getAddressCount using bigInts to prevent IPv6 overflow (needed for the divide implementation)
* Added getNextAddress functionality  and the "cidr next" command (also needed for the divide implementation)
* Added "cidr divide" functionality
* Clarified the documentation of the getUseableAddresses command, because it gets all the addresses....not just the usable hosts.
* Added some unit tests. nothing crazy.

Closes #89

### CODE REVIEW

**Comment by bschaatsbergen** [comment]
Hey @Phaze228, thank you very much for submitting this pull request. Give me a couple working days (I'll be out over the weekend) to review this. Your contribution is much appreciated! 👏🏼

**Comment by Phaze228** [comment]
Not a problem, I may clean up a few things, over the weekend. Notably, missing the core_test.go file was testing the AddressCount function, and a few other little things the auto-analyzer is yelling at. But functionally, it seems good.
Probably have to come up with a nice way to handle dividing ipv6 spaces, because if you want to do something stupid like dividing a 2001::0000/4 into 90,000,000 different spaces, that all gets printed to the console.

**Comment by Phaze228** [comment]
I will say, it's a little rude of the linter to point out my mistakes without files and line numbers.

**Comment by bschaatsbergen** [comment]
> I will say, it's a little rude of the linter to point out my mistakes without files and line numbers.

Hmm agreed, I'll take a look today at whether we can fix this.

**Comment by Phaze228** [comment]
Beautiful. Thank you.

File: README.md
Comment (bschaatsbergen) [inline README.md:101]:
```suggestion
To divide a CIDR range into N distinct networks:
```

File: README.md
Comment (bschaatsbergen) [inline README.md:133]:
```suggestion
You can also use the `-u` flag to divide the network based on the number of desired host addresses. This calculation includes a broadcast and gateway address, so you only need to consider the host addresses.
```

Comment (bschaatsbergen) [reply README.md:133]: I also suggest we change from the `-u` flag to something more related to "host addresses" since that's a more commonly accepted term. I can't quickly think of a good name for the flag, allow me to get back on this please.

File: README.md
Comment (bschaatsbergen) [inline README.md:134]:
```suggestion
The command below shows how to divide a CIDR range to accommodate a minimum of 32 hosts, 30 hosts, and 12 hosts. It also provides the total possible hosts available for a subnet.
```

File: cmd/count.go
Comment (bschaatsbergen) [inline cmd/count.go:52]: Good spot!

File: cmd/divide.go
Comment (bschaatsbergen) [inline cmd/divide.go:22]: If possible, let's rename "Users' to "Hosts' in this file? It's a more common accepted term that folks grasp easier.

File: cmd/explain.go
Comment (bschaatsbergen) [inline cmd/explain.go:102]: Why are we doing this exactly? Is it not possible to print always as a string for consistency otherwise?

File: cmd/divide.go
Comment (bschaatsbergen) [inline cmd/divide.go:147]: Do you think some of this logic should go to Core? Most logic is now in Core to perform such operations.

File: cmd/next.go
Comment (bschaatsbergen) [inline cmd/next.go:19]: I really like the idea of this. Unfortunately I have no access to your repository to clone and test it. Would it be possible to get access to your repo for me to test this command?

File: README.md
Comment (Phaze228) [reply README.md:133]: Oh I agree. My original desire was -h and --hosts. But cobra's default and the other standard is -h for help.
-d for domains?
-n for numbered hosts?
-c for clients (which is probably more technically accurate than users)?
but -u was for users was just the fastest sensible alternative when I realized -h failed.

File: cmd/explain.go
Comment (Phaze228) [reply cmd/explain.go:102]: I tried to preserve the earlier commits of keeping the comma's in the printed string (because it does look nice).
Eg. ```123,456``` vs. ```123456``` 
 NewPrinter does that, but it doesn't seem to interpret *big.Int as an interpret-able number so that's the quick fix to preserve comma printing for reasonably sized numbers. 
I could probably carve out a simple formatter, but I didn't know how much hassle it would be, so I opted to just preserve the earlier function first before trying to add in extra niceties.

File: cmd/explain.go
Comment (bschaatsbergen) [inline cmd/explain.go:99]: FormtNumber seems undefined

File: cmd/divide.go
Comment (bschaatsbergen) [inline cmd/divide.go:60]: Perhaps we can move to Int64Slice, instead of Int64SliceP to force a user to use `--hosts` instead of `-H` which might confuse them as `-h` is for help

**Comment by bschaatsbergen** [comment]
I'm very excited to get this merged, thanks for your hard work. It's just a few minor tweaks we need to make.

File: cmd/explain.go
Comment (Phaze228) [reply cmd/explain.go:99]: That's my  bad, been git committing -a, forgot it doesn't factor in NEW files. I'll push that real quick.

File: cmd/divide.go
Comment (Phaze228) [reply cmd/divide.go:60]: I'll let you have that call, as I am a big fan of shorthands for CLI tools. When they're thrown in scripts, that's when I prefer the longer syntax.

**Comment by bschaatsbergen** [comment]
@mmourick requesting your review too 👍🏼

File: pkg/core/core.go
Comment (mmourick) [inline pkg/core/core.go:231]: Is `Wack` an acronym of some sorts? Wouldn't a name like `cidrMask` be more clear?

File: README.md
Comment (mmourick) [inline README.md:96]: Looks like this line accidentally got removed
```suggestion
true
```

File: cmd/explain.go
Comment (mmourick) [inline cmd/explain.go:100]: Please remove the commented out code

**Review by mmourick — COMMENTED** [review]
Thank you for your contribution! This looks like a great addition to `cidr`!

I've left a few small nitpicks, but besides that I think it's almost good to go. Just need to run `gofmt` as currently `golangci-lint` is failing.

File: pkg/core/core.go
Comment (Phaze228) [reply pkg/core/core.go:231]: The "wack" is technically a "/" when talking about ip's and cidr masks verbally.
10.0.0.0 "wack" 24.  Yes, it's a bit a silly to use it as a variable name, but it's where my mind went. But yeah we can fix it, it's an easy change.

**Comment by Phaze228** [comment]
Took in everyone's suggestions. Minus removing the shortened -H for hosts flag. My preference is for it, but you guys can adjust it if you want before merging. 
Hopefully the linter doesn't continue to hate me.

**Comment by bschaatsbergen** [comment]
Hey @Phaze228, update: I'm working locally on your branch 👍🏼

**Comment by Phaze228** [comment]
Alrighty, sounds good. Let me know if you need anything.

**Comment by bschaatsbergen** [comment]
Hey @Phaze228, apologies for the slower review cycle. Work has been a bit hectic lately. I trimmed down this pull request to just include the `divide` command to make the review process easier. I’m very keen on getting the `next` subcommand in, but I’d like to review and submit that separately through a follow-up PR.

**Review by bschaatsbergen — DISMISSED** [review]
LGTM 🚀

**Comment by bschaatsbergen** [comment]
> Could you please elaborate on one thing? I’m curious about why we introduced a format function and what issues were found with the existing formatting. The previous implementation seemed less complex and didn’t have any issues, as far as I can recall.

It appears that the `NewPrinter` from `golang.org/x/text/language` cannot handle a `*big.Int`.

**Review by bschaatsbergen — APPROVED** [review]
LGTM 🚀

**Comment by bschaatsbergen** [comment]
Thank you for your hard work on this, @Phaze228! It’s greatly appreciated 👏🏼!

**Pull request merged**
